SharedLibrary/Sources/Shared/Grid.inl

```diff
@@ -0,0 +1,80 @@
+template<typename T>
+void Grid<T>::SetSize(vec3u size) {
+	dataSize = size;
+	data.resize(size.x*size.y*size.z);
+}
+
+template<typename T>
+void Grid<T>::SetSize(uint x, uint y, uint z) {
+	SetSize({x,y,z});
+}
+
+template<typename T>
+vec3u Grid<T>::GetSize() const {
+	return dataSize;
+}
+
+template<typename T>
+std::vector<T>& Grid<T>::Get() {
+	return data;
+}
+
+template<typename T>
+const std::vector<T>& Grid<T>::Get() const {
+	return data;
+}
+
+template<typename T>
+typename std::vector<T>::reference Grid<T>::At(vec3u pos) {
+	return data[flat_index(pos, dataSize.x, dataSize.y)];
```

Comment:
We wanted to remove flat_index and use Grid instead.

I suggest move this function from global scope to Grid as private member.